src/libraries/System.Net.Http.WinHttpHandler/tests/FunctionalTests/WinHttpHandlerTest.cs

```diff
@@ -69,6 +69,72 @@ public async Task GetAsync_ConcurrentRead_ThrowsInvalidOperationException()
             await Task.WhenAll(tasks);
         }
 
+        [OuterLoop]
+        [ConditionalFact(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
+        public async Task SendAsync_ServerCertificateValidationCallback_CalledOnce()
+        {
+            await RemoteExecutor.Invoke(async () =>
+            {
+                AppContext.SetSwitch("System.Net.Http.UseWinHttpCertificateCaching", true);
+                int callbackCount = 0;
+                var handler = new WinHttpHandler()
+                {
+                    ServerCertificateValidationCallback = (_, _, _, _) =>
+                    {
+                        Interlocked.Increment(ref callbackCount);
+                        return true;
+                    }
+                };
+                using (var client = new HttpClient(handler))
+                {
+                    for (int i = 0; i < 5; i++)
+                    {
+                        var response = await client.SendAsync(new HttpRequestMessage(HttpMethod.Get, Configuration.Http.SecureRemoteEchoServer)
+                        {
+                            Version = HttpVersion.Version11
+                        });
+                        Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+                        _ = await response.Content.ReadAsStringAsync();
+                    }
+                    Assert.Equal(1, callbackCount);
+                }
+                AppContext.SetSwitch("System.Net.Http.UseWinHttpCertificateCaching", false);
+            }).DisposeAsync();
+        }
+
+        [OuterLoop]
+        [ConditionalFact(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
+        public async Task SendAsync_ServerCertificateValidationCallbackHttp2_CalledOnce()
```

Comment:
Nit: Can we make the version a theory parameter instead of duplicating the test for 1.1 and 2.0?